LEAF_Request_Portal/js/workflow.js

```diff
@@ -45,29 +45,51 @@ var LeafWorkflow = function(containerID, CSRFToken) {
             antiDblClick = 1;
         }
 
+        // Check if CSRFToken has Changed (Timeout Fix)
+        $.ajax({
+            type: 'GET',
+            url: rootURL + 'api/?a=formWorkflow/getCSRFToken',
+            async: false,
+            success: function(res) {
+                data.CSRFToken = res;
+            },
+            error: function(err) {
+                alert('Session Expired');
+            }
+        });
+
         $("#workflowbox_dep" + data['dependencyID']).html('<div style="border: 2px solid black; text-align: center; font-size: 24px; font-weight: bold; background: white; padding: 16px; width: 95%">Applying action... <img src="'+ rootURL +'images/largespinner.gif" alt="loading..." /></div>');
         $.ajax({
             type: 'POST',
             url: rootURL + 'api/?a=formWorkflow/' + currRecordID + '/apply',
             data: data,
             success: function(response) {
-                if(response.errors.length == 0) {
-                    $("#workflowbox_dep" + data['dependencyID']).html('<div style="border: 2px solid black; text-align: center; font-size: 24px; font-weight: bold; background: white; padding: 16px; width: 95%">Action applied!</div>');
-                    $("#workflowbox_dep" + data['dependencyID']).hide('blind', 500);
+                if (response !== "Invalid Token.") {
+                    if (response.errors.length === 0) {
+                        $("#workflowbox_dep" + data['dependencyID']).html('<div style="border: 2px solid black; text-align: center; font-size: 24px; font-weight: bold; background: white; padding: 16px; width: 95%">Action applied!</div>');
+                        $("#workflowbox_dep" + data['dependencyID']).hide('blind', 500);
 
-                    getWorkflow(currRecordID);
-                    if(actionSuccessCallback != undefined) {
-                        actionSuccessCallback();
-                    }
-                }
-                else {
-                    var errors = '';
-                    for(var i in response.errors) {
-                        errors += response.errors[i] + '<br />';
+                        getWorkflow(currRecordID);
+                        if (actionSuccessCallback !== undefined) {
+                            actionSuccessCallback();
+                        }
+                    } else {
+                        var errors = '';
+                        for (var i in response.errors) {
+                            errors += response.errors[i] + '<br />';
```

Comment:
Not going to through a flag on this at this time.  But concatenation inside a loop is generally a bad idea.  The elements should be stored in an array than then imploded once the looping is done.